### PR TITLE
Bunch more comments and docstrings in cedar-policy-validator

### PR DESCRIPTION
**Pull request by cdisselkoen**
## Description of changes

Add more comments and docstrings through `cedar-policy-validator`.

Brings `cedar-policy-validator` to the same warning/clippy settings we use for `cedar-policy`.

Many of these changes are unremarkable, but careful review is requested for some of the invariants/assumptions newly documented in `namespace_def.rs`.

No functional changes.

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

### CODE REVIEW

File: cedar-policy-validator/src/lib.rs [hunk 1]
```diff
@@ -16,7 +16,20 @@
 
 //! Validator for Cedar policies
 #![forbid(unsafe_code)]
+#![warn(rust_2018_idioms, clippy::pedantic, clippy::nursery)]
+#![deny(
+    missing_docs,
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/lib.rs:21]: Is this worth the effort for internal crates?

File: cedar-policy-validator/src/schema/namespace_def.rs [hunk 2]
```diff
@@ -64,80 +64,131 @@ pub(crate) fn is_action_entity_type(ty: &Name) -> bool {
...
     pub(super) applies_to: ValidatorApplySpec,
     /// The direct parent action entities for this action.
+    /// These are fully qualified `EntityUID`s, but may be actions declared in a
+    /// different namespace or schema fragment, and thus not declared yet.
+    /// We will check for undeclared parents when combining fragments into a
+    /// `ValidatorSchema`.
     pub(super) parents: HashSet<EntityUID>,
     /// The types for the attributes defined for this actions entity.
+    /// Although these are not literally wrapped in `WithUnresolvedTypeDefs`,
+    /// they are in spirit: they may refer to common types which have not yet
```
Comment (john-h-kastner-aws) [inline cedar-policy-validator/src/schema/namespace_def.rs:185]: I don't think this is accurate. These should be fully specified.

(we should honestly just rip out action attributes, they're not in any public api)

**Review by john-h-kastner-aws — APPROVED** [review]

File: cedar-policy-validator/src/lib.rs (hunk 1 above)
Comment (cdisselkoen) [reply cedar-policy-validator/src/lib.rs:21]: Maybe not in retrospect, but now that this PR added the doc comments to bring us into compliance, it's probably worth to maintain?

File: cedar-policy-validator/src/schema/namespace_def.rs (hunk 2 above)
Comment (cdisselkoen) [reply cedar-policy-validator/src/schema/namespace_def.rs]: Fixed the comment. Thanks for the review.

**Review by shaobo-he-aws — APPROVED** [review]

**Review by shaobo-he-aws — APPROVED** [review]

File: cedar-policy-validator/src/lib.rs (hunk 1 above)
Comment (john-h-kastner-aws) [reply cedar-policy-validator/src/lib.rs:21]: Yeah, fine to keep around

**Review by john-h-kastner-aws — APPROVED** [review]

**Pull request merged**